DNS switch to new net_context API

tbursztyka opened the pull request.

Commit:
net/dns: Move common macros in the relevant header

Some of dns_pack.c macros are also used in resolve.c, so let's just put
the macros in dns_pack.h

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/dns: Move function declaration after macros and local variables

dns_write was in the middle of this, let's just move it to a more
relevant place so we logically have macros, then local variables then
functions.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: Add a way to set/get ttl/hop_limit from net_context

Since net_context_sendto_new() does not take a net_pkt anymore, the only
way to set net_pkt's ttl/hop_limit is to pass it through net_context.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/dns: Switch to new net_context API

So instead of building the net_pkt by itself, dns will pass its message
buffer to net_context_sendto_new(), which in turn will build the net_pkt
relevantly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
mdns will follow shortly, so no hurry to apply that PR now.

jukkar commented on include/net/net_context.h:
```diff
@@ -293,6 +293,15 @@ struct net_context {
 	/** Network interface assigned to this context */
 	s8_t iface;
+
+
```
extra empty line

jukkar commented on include/net/net_context.h:
```diff
@@ -293,6 +293,15 @@ struct net_context {
 	s8_t iface;
+
+
+	/** IPv6 hop limit or IPv4 ttl for this network packet.
```
Do we really need to store this in net_context? The hop limit/ttl is already stored in net_if and we have interface index in net_context so we could fetch it from net_if directly.

tbursztyka replied:
What about temporary ttl/hop_limit tweak? As far as I can see, that's what is happening in resolve.c

jukkar replied:
Yep, you are right here, I missed the temp hop-limit setting in LLMNR

tbursztyka replied:
Indeed, quickly  c/p from net_pkt

Commit:
net/dns: Switch llmnr responder to new net_pkt API

Unfortunately, most of the dns pack/unpack code is not factorized, and
llmnr responder rewrites its own functions to write dns hdr, query,
answer.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/dns: Switch mdns responder to new net_pkt API

Unfortunately, most of the dns pack/unpack code is not factorized, and
mdns responder rewrites its own functions to write dns hdr, query,
answer.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
Note that I still need to test llmnr and mdns responder

pfalcon reviewed (dismissed):
Minor issues spotted so far, still reviewing.

jukkar reviewed (approved):
LGTM

pfalcon reviewed (approved):
Based on the discussion in https://github.com/zephyrproject-rtos/zephyr/pull/13506, there's little choice but just approve it.
